src/python/pants/backend/experimental/python/lockfile_test.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from unittest.mock import MagicMock
+
+from pants.backend.experimental.python.lockfile import (
+    PythonLockfileRequest,
+    lockfile_metadata,
+    read_lockfile_metadata,
+    validated_lockfile_content,
+)
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+def test_metadata_round_trip():
```

Comment:
This is annoying, but: to get MyPy to type check the body of this function, the signature needs a type hint.

```suggestion
def test_metadata_round_trip() -> None:
```

Ditto to the rest of the file.